workshop/content/exercises/04-remediations.md

```diff
@@ -1,9 +1,391 @@
+
+```
+$ oc get machineconfigpools worker -o yaml | grep "\spaused"
+  paused: true
+```
+
+Now that the pool is paused, we can freely apply a couple of remediations and be sure that there won't
+be any unnecessary node reboots.
+
+Let's do so!
+
+```
+$ oc patch complianceremediations rhcos4-e8-worker-sysctl-kernel-kexec-load-disabled \
+    -p '{"spec":{"apply":true}}' --type=merge
+complianceremediation.compliance.openshift.io/rhcos4-e8-worker-sysctl-kernel-kexec-load-disabled patched
+$ oc patch complianceremediations rhcos4-e8-worker-sysctl-kernel-unprivileged-bpf-disabled \
+    -p '{"spec":{"apply":true}}' --type=merge
+complianceremediation.compliance.openshift.io/rhcos4-e8-worker-sysctl-kernel-unprivileged-bpf-disabled patched
+```
+
+We can now browse the **MachineConfig** object and check that the new remediations have been included.
+
+```
+$ oc get machineconfig 75-rhcos4-e8-worker-periodic-e8 -o jsonpath='{.spec.config.storage.files}' | jq
+[
+  {
+    "contents": {
+      "source": "data:,kernel.dmesg_restrict%3D1"
+    },
+    "filesystem": "root",
+    "mode": 420,
+    "path": "/etc/sysctl.d/75-sysctl_kernel_dmesg_restrict.conf"
+  },
+  {
+    "contents": {
+      "source": "data:,kernel.kexec_load_disabled%3D1"
+    },
+    "filesystem": "root",
+    "mode": 420,
+    "path": "/etc/sysctl.d/75-sysctl_kernel_kexec_load_disabled.conf"
+  },
+  {
+    "contents": {
+      "source": "data:,kernel.unprivileged_bpf_disabled%3D1"
+    },
+    "filesystem": "root",
+    "mode": 420,
+    "path": "/etc/sysctl.d/75-sysctl_kernel_unprivileged_bpf_disabled.conf"
+  }
+]
+```
+
+Now that the compliance-operator has persisted the remediations to the relevant object,
+we may un-pause the pool and let the machine-config-operator persist the changes.
+
+```
+$ oc patch machineconfigpools worker -p '{"spec":{"paused":false}}' --type=merge
+machineconfigpool.machineconfiguration.openshift.io/worker patched
+```
+
+### To what *MachineConfigPool* will my remediations be applied to?
+
+You might have noticed that in this case that these remediations applied to only the `worker`
+**MachineConfigPool**. This is because the **ComplianceSuite** created two **ComplianceScan** objects
+under the hood. These scans have a specific `nodeSelector` set that should match a **MachineConfigPool**. 
+This way we ensure that the remediations are applicable to nodes with similar configurations, thus 
+reducing inconsistencies and helping us keep the remediations applicable. This was all abstracted
+away thanks to the `roles` key in the **ScanSettings** object.
+
+Letting the operator apply the remediations
+-------------------------------------------
+
+Applying each scan can be fairly tedious. This is why the compliance-operator has a flag for this!
+
+
+Applying scans is done in a suite level, which means it's done for all the scans that belong to the 
+benchmark you're currently scanning.
```

Comment:
Should we suggest only doing this once the remediations have been reviewed or audited?

Reply:
Done.